File: android/app/src/main/kotlin/com/wrbug/gitbbs/flutterbridge/mmkv/MmkvFlutterChannel.kt
package com.wrbug.gitbbs.flutterbridge.mmkv

import com.wrbug.gitbbs.flutterbridge.FlutterChannel
import com.wrbug.gitbbs.flutterbridge.FlutterMethodCall
import com.wrbug.gitbbs.flutterbridge.mmkv.method.GetTokenCall
import com.wrbug.gitbbs.flutterbridge.mmkv.method.GetUserCall
import com.wrbug.gitbbs.flutterbridge.mmkv.method.SaveTokenCall
import com.wrbug.gitbbs.flutterbridge.mmkv.method.SaveUserCall
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import java.util.HashMap

class MmkvFlutterChannel : FlutterChannel() {
    val calls: Map<String, FlutterMethodCall> by lazy {
        val map = HashMap<String, FlutterMethodCall>()
        var call: FlutterMethodCall = GetTokenCall()
        map[call.getMethodName()] = call
        call = SaveTokenCall()
        map[call.getMethodName()] = call
        call = GetUserCall()
        map[call.getMethodName()] = call
        call = SaveUserCall()
        map[call.getMethodName()] = call
        map
    }

    override fun dispatch(call: MethodCall, result: MethodChannel.Result) {
        if (calls.containsKey(call.method).not()) {
            result.notImplemented()
            return
        }
        calls[call.method]?.apply {
            invoke(call, result)
        }
    }

    override fun getChannel() = "mmkv"
}

File: ios/Runner/FlutterBridge/mmkv/MmkvFlutterChannel.swift
//
//  MmkvFlutterChannel.swift
//  Runner
//
//  Created by WrBug on 2019/3/7.
//  Copyright © 2019 The Chromium Authors. All rights reserved.
//

import Foundation
class  MmkvFlutterChannel: FlutterChannel {
    var calls=[String:MethodCall]();
    override init() {
        var call:MethodCall=GetUserCall.init()
        calls[call.getMethodName()]=call
         call=SaveUserCall.init()
        calls[call.getMethodName()]=call
         call=GetTokenCall.init()
        calls[call.getMethodName()]=call
         call=SaveTokenCall.init()
        calls[call.getMethodName()]=call
    }
    override func getChannel() -> String {
        return "mmkv"
    }
    override func dispatch(_ call: FlutterMethodCall, _ result: (Any?) -> Void) {
        let methodCall=calls[call.method]
        if(methodCall != nil){
            methodCall?.invoke(call: call, result: result)
            return
        }
        result(FlutterMethodNotImplemented)
    }
}
